intelij-plugin/src/main/java/tech/pantheon/yanginator/plugin/external/ExternalRules.java

```diff
@@ -96,4 +94,94 @@ private static boolean unanimousResults(boolean[] listOfBooleanResults) {
         }
         return true;
     }
+
+    /**
+     * For loop extracts hex ranges with characters that are forbidden and add those characters into charList,
+     * in addition char with hex code 0xFFFD is added manually even though it's not forbidden. This is because
+     * idea can't recognize some characters from forbidden ranges and replace them with this value.
+     * Second for loop extracts current yang-string from original text into variable text.
+     * Last for loop compares string characters with forbidden characters and stops at 1st match.
+     *
+     * @param psiBuilder Psi builder
+     * @param level      Level
+     * @param parser     parser for parsing current string
+     * @param strings    list of strings containing forbidden ranges for character
+     * @return return true if forbidden range doesn't contain any character of this string
+     */
+    public static boolean checkString(PsiBuilder psiBuilder, int level, Parser parser, String... strings) {
+        if (!recursion_guard_(psiBuilder, level, "rule")) return false;
+
+        final List<String> stringList = new ArrayList<>(List.of(strings));
+        final List<Character> charList = new ArrayList<>();
+        final Set<Character> separators = new HashSet<>(Set.of('\n', '\r', ' ', '\t'));
+
+        // 0xFFFD is for characters that can't be recognized by idea and are replaced by this value
+        charList.add((char) 0xFFFD);
+        for (String s : stringList) {
```

Comment:
looks like duplicate code, please extract this to private static method to remove duplications

Reply:
Done.